src/base.js

```diff
@@ -17,16 +20,24 @@ class Base extends EventEmitter {
   constructor(...params) {
     super();
 
-    if (params[0] && params[0].tokenId) {
+    if (params[0] instanceof Base) {
       return Object.assign(this, params[0]);
     }
 
-    this.tokenId = params[0] || process.env.MUX_TOKEN_ID;
-    this.tokenSecret = params[1] || process.env.MUX_TOKEN_SECRET;
+    if (typeof params[0] === 'object') {
+      this.config = params[0]; // eslint-disable-line prefer-destructuring
+      this.tokenId = undefined;
+      this.tokenSecret = undefined;
+    } else {
+      this.config = {};
+      this.tokenId = params[0]; // eslint-disable-line prefer-destructuring
+      this.tokenSecret = params[1]; // eslint-disable-line prefer-destructuring
+    }
 
     this.http = axios.create({
-      baseURL: 'https://api.mux.com',
+      baseURL: this.config.baseUrl,
```

Comment:
this isn't always going to exist, especially in the case that you don't pass any config on at all